### PR TITLE
Add reources management for nodes

### PR DESCRIPTION
**Pull request by puppetninja**

### CODE REVIEW

File: charts/tezos/templates/_containers.tpl [hunk 1]
```diff
@@ -43,6 +43,7 @@
      *    with_config    bring in the configMap defaults true only on utils.
      *    with_secret    bring in the secrets map including the identities.
      *    localvars      set env vars MY_* Defaults to true only on utils.
+     *    resources      set resources management to be empty.
```
Comment (nicolasochem) [inline charts/tezos/templates/_containers.tpl:46]: unclear what this means

File: charts/tezos/templates/_containers.tpl [hunk 2]
```diff
@@ -238,9 +246,11 @@
 
 {{- define "tezos.container.sidecar" }}
   {{- if or (not (hasKey $.node_vals "readiness_probe")) $.node_vals.readiness_probe }}
-    {{- include "tezos.generic_container" (dict "root"  $
-                                                "type"  "sidecar"
-                                                "image" "utils"
+    {{- $sidecarResources := dict "requests" (dict "memory" "150Mi") "limits" (dict "memory" "200Mi") -}}
```
Comment (nicolasochem) [inline charts/tezos/templates/_containers.tpl:249]: sidecar resources and limits are hardcoded? why are they necessary?

File: charts/tezos/templates/_containers.tpl [hunk 3]
```diff
@@ -260,24 +270,45 @@
 
 {{- define "tezos.container.node" }}
 {{- if eq (include "tezos.getNodeImplementation" $) "octez" }}
-    {{- include "tezos.generic_container" (dict "root"        $
-                                                "type"        "octez-node"
-                                                "image"       "octez"
-                                                "with_config" 0
-    ) | nindent 0 }}
+    {{- if hasKey $.node_vals "resources" }}
```
Comment (nicolasochem) [inline charts/tezos/templates/_containers.tpl:273]: there may be a more compact way to ahcieve this? if resources are undefined, then omit these settings in the final object returned by the function

**Review by nicolasochem — COMMENTED** [review]
Left some inline questions. Also, it would be good to "document" this in the comments of `charts/tezos/values.yaml`.

File: charts/tezos/templates/_containers.tpl (hunk 3 above)
Comment (puppetninja) [reply charts/tezos/templates/_containers.tpl:273]: yes, I will work on this

File: charts/tezos/templates/_containers.tpl (hunk 2 above)
Comment (puppetninja) [reply charts/tezos/templates/_containers.tpl:249]: I think user has no overview of resource needs for sidecar so we should hard code it here as when user doesn't specify the readiness_probe we inject this side automatically to replace it

**Comment by puppetninja** [comment]
opened #508

**Pull request closed without merging**